rpc/auth.go

```diff
@@ -13,35 +13,49 @@ import (
 	"github.com/golang-jwt/jwt/v4"
```

Comment:
I'd say this file is where all the interesting API changes are happening. We don't have this module versioned so we are free to break things. I'd call this v2 of rpc auth with respect to the API. Using this feels much better to me now and makes more sense to me.